functests/5_latency_testing/latency_testing.go

```diff
@@ -88,57 +93,96 @@ type latencyTest struct {
+					Skip("The executable test file does not exist , skipping the test.")
+				}
+				output, err := exec.Command("../../build/_output/bin/latency-e2e.test", "-ginkgo.focus", test.toolToTest).Output()
+				if err != nil {
+					//we don't log Error level here because the test might be a negative check
+					testlog.Info(err.Error())
+				}
+
+				ok, matchErr := regexp.MatchString(skipInsufficientCpu, string(output))
+				if matchErr != nil {
+					testlog.Error(matchErr.Error())
+				}
+				if ok {
+					testlog.Info(skipInsufficientCpu)
+					continue
+				}
+
+				if isPositiveTest {
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+					Expect(string(output)).NotTo(MatchRegexp(unexpectedError), "Unexpected error was detected in a positive test")
+					//Check runtime argument in the pod's log only if the tool is expected to be executed
+					ok, matchErr := regexp.MatchString(success, string(output))
+					if matchErr != nil {
+						testlog.Error(matchErr.Error())
+					}
+					if ok {
+						commandRegex := fmt.Sprintf("%s command with arguments .*--duration %s", test.toolToTest, test.testRuntime)
+						Expect(string(output)).To(MatchRegexp(commandRegex), "The output of the executed tool is not as expected")
+					}
+				}
+				for _, msg := range test.outputMsgs {
+					Expect(string(output)).To(MatchRegexp(msg), "The output of the executed tool is not as expected")
+				}
+			}
+		},
+			table.Entry("[test_id:42851] Latency tools shouldn't run with default environment variables values", []latencyTest{{outputMsgs: []string{skip, skipTestRun}}}, positiveTesting),
+			table.Entry("[test_id:42850] Oslat - Verify that the tool is working properly with valid environment variables values", getValidValuesTests(oslat), positiveTesting),
+			table.Entry("[test_id:42853] Oslat - Verify that the latency tool test should print an expected error message when passing invalid environment variables values", getNegativeTests(oslat), negativeTesting),
+			table.Entry("[test_id:42115] Cyclictest - Verify that the tool is working properly with valid environment variables values", getValidValuesTests(cyclictest), positiveTesting),
+			table.Entry("[test_id:42852] Cyclictest - Verify that the latency tool test should print an expected error message when passing invalid environment variables values", getNegativeTests(cyclictest), negativeTesting),
+			table.Entry("[test_id:42849] Hwlatdetect - Verify that the tool is working properly with valid environment variables values", getValidValuesTests(hwlatdetect), positiveTesting),
+			table.Entry("[test_id:42856] Hwlatdetect - Verify that the latency tool test should print an expected error message when passing invalid environment variables values", getNegativeTests(hwlatdetect), negativeTesting),
+		)
+	})
+
+	Context("Verify tests output with existed performance profile", func() {
+		It("[test_id:---] should skip oslat test when isolated CPUs amount is less than 2", func() {
```

Comment:
blank test id ?

Reply:
not intended, will fix it, thanks for catching that!

Reply:
not intended, will fix it, thanks for catching that!